core/templates/dev/head/pages/exploration_editor/editor_tab/state_editor_solution.html

```diff
@@ -0,0 +1,128 @@
+<div ng-controller="StateSolution">
+  <div ng-if="!isCurrentInteractionLinear() &&
+              stateHintsService.displayed.length > 0 &&
+              objectType !== 'LogicQuestion'">
```

Comment:
Don't use specific interaction IDs in the main code, which is supposed to be general in nature.

At worst, have a constant along the lines of INTERACTION_IDS_WITH_NO_SOLUTION_DISPLAY = [...] (I'm not sure about the naming but you get the idea) and refer to that. Include enough explanation in the codebase so that devs know what's going on and why some interactions are excluded.